File: oolab/src/test/java/agh/ics/oop/model/MapDirectionTest.java
package agh.ics.oop.model;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertSame;

public class MapDirectionTest {
    @Test
    public void addTwoTest() {
        assertSame(MapDirection.NORTH.add(2), MapDirection.EAST);
        assertSame(MapDirection.EAST.add(2), MapDirection.SOUTH);
        assertSame(MapDirection.SOUTH.add(2), MapDirection.WEST);
        assertSame(MapDirection.WEST.add(2), MapDirection.NORTH);
    }

    @Test
    public void addTest() {
        assertSame(MapDirection.NORTH.add(6), MapDirection.WEST);
        assertSame(MapDirection.WEST.add(3), MapDirection.NORTHEAST);
        assertSame(MapDirection.SOUTHEAST.add(7), MapDirection.EAST);
        assertSame(MapDirection.SOUTHWEST.add(5), MapDirection.EAST);
        assertSame(MapDirection.NORTHWEST.add(2), MapDirection.NORTHEAST);
    }
}

File: kotlinlab/src/test/kotlin/model/MapDirectionTest.kt
package model

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe

class MapDirectionTest : FunSpec({

    test("nextTest") {
        MapDirection.NORTH.next() shouldBe MapDirection.EAST
        MapDirection.WEST.next() shouldBe MapDirection.NORTH
        MapDirection.SOUTH.next() shouldBe MapDirection.WEST
        MapDirection.EAST.next() shouldBe MapDirection.SOUTH
    }

    test("previousTest") {
        MapDirection.NORTH.previous() shouldBe MapDirection.WEST
        MapDirection.WEST.previous() shouldBe MapDirection.SOUTH
        MapDirection.SOUTH.previous() shouldBe MapDirection.EAST
        MapDirection.EAST.previous() shouldBe MapDirection.NORTH
    }
})